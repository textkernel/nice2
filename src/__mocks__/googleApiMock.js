/* eslint-disable class-methods-use-this */
// This is an incolplete stubbing of google API, based on bits and pieces.
// Feel free to extend it as required.
const noop = () => {};
const EMPTY_OBJECT = {};
const EMPTY_CLASS = class {};

export const fitBoundsMock = jest.fn();
export const setZoomMock = jest.fn();
export const setCenterMock = jest.fn();
export const getPlacePredictionsMock = jest.fn();

export class Map {
    fitBounds() {
        fitBoundsMock();
    }

    getBounds() {}

    getCenter() {}

    getDiv() {}

    getHeading() {}

    getMapTypeId() {}

    getProjection() {}

    getStreetView() {}

    getTilt() {}

    getZoom() {}

    panBy() {}

    panTo() {}

    panToBounds() {}

    setCenter() {
        setCenterMock();
    }

    setHeading() {}

    setMapTypeId() {}

    setOptions() {}

    setStreetView() {}

    setTilt() {}

    setZoom() {
        setZoomMock();
    }

    controls = {};

    data = {
        add: noop,
        addGeoJson: noop,
        contains: noop,
        forEach: noop,
        getControlPosition: noop,
        getControls: noop,
        getDrawingMode: noop,
        getFeatureById: noop,
        getMap: noop,
        getStyle: noop,
        loadGeoJson: noop,
        overrideStyle: noop,
        remove: noop,
        revertStyle: noop,
        setControlPosition: noop,
        setControls: noop,
        setDrawingMode: noop,
        setMap: noop,
        setStyle: noop,
        toGeoJson: noop,
        controlPosition: EMPTY_OBJECT,
        controls: [],
    };

    mapTypes = {
        set: noop,
    };

    overlayMapTypes = {
        clear: noop,
        getArray: noop,
        getAt: noop,
        getLength: noop,
        insertAt: noop,
        pop: noop,
        push: noop,
        removeAt: noop,
        setAt: noop,
    };
}

export class Marker {
    getAnimation() {}

    getClickable() {}

    getCursor() {}

    getDraggable() {}

    getIcon() {}

    getLabel() {}

    getMap() {}

    getOpacity() {}

    getPosition() {}

    getShape() {}

    getTitle() {}

    getVisible() {}

    getZIndex() {}

    setAnimation() {}

    setClickable() {}

    setCursor() {}

    setDraggable() {}

    setIcon() {}

    setLabel() {}

    setMap() {}

    setOpacity() {}

    setOptions() {}

    setPosition() {}

    setShape() {}

    setTitle() {}

    setVisible() {}
}

export class LatLng {
    equals() {}

    lat() {}

    lng() {}

    toJSON() {}

    toString() {}

    toUrlValue() {}
}

export class LatLngBounds {
    contains() {}

    equals() {}

    extend() {}

    getCenter() {}

    getNorthEast() {}

    getSouthWest() {}

    intersects() {}

    isEmpty() {}

    toJSON() {}

    toSpan() {}

    toString() {}

    toUrlValue() {}

    union() {}
}

export class Circle {
    getBounds() {}

    getCenter() {}

    getDraggable() {}

    getEditable() {}

    getMap() {}

    getRadius() {}

    getVisible() {}

    setCenter() {}

    setDraggable() {}

    setEditable() {}

    setMap() {}

    setOptions() {}

    setRadius() {}

    setVisible() {}
}

export const google = {
    maps: {
        Animation: EMPTY_CLASS,
        Attribution: EMPTY_CLASS,
        BicyclingLayer: EMPTY_CLASS,
        Circle,
        ControlPosition: EMPTY_CLASS,
        Data: EMPTY_CLASS,
        DirectionsRenderer: EMPTY_CLASS,
        DirectionsService: EMPTY_CLASS,
        DirectionsStatus: EMPTY_CLASS,
        DistanceMatrixElementStatus: EMPTY_CLASS,
        DistanceMatrixService: EMPTY_CLASS,
        DistanceMatrixStatus: EMPTY_CLASS,
        ElevationService: EMPTY_CLASS,
        ElevationStatus: EMPTY_CLASS,
        FusionTablesLayer: EMPTY_CLASS,
        GeocoderLocationType: EMPTY_CLASS,
        GroundOverlay: EMPTY_CLASS,
        ImageMapType: EMPTY_CLASS,
        InfoWindow: EMPTY_CLASS,
        KmlLayer: EMPTY_CLASS,
        KmlLayerStatus: EMPTY_CLASS,
        LatLng,
        LatLngBounds,
        Map,
        Marker,
        MapTypeControlStyle: EMPTY_CLASS,
        MapTypeId: {
            HYBRID: '',
            ROADMAP: '',
            SATELLITE: '',
            TERRAIN: '',
        },
        MapTypeRegistry: EMPTY_CLASS,
        MarkerImage: EMPTY_CLASS,
        MarkerOptions: EMPTY_CLASS,
        MaxZoomService: EMPTY_CLASS,
        MaxZoomStatus: EMPTY_CLASS,
        MVCArray: EMPTY_CLASS,
        MVCObject: EMPTY_CLASS,
        NavigationControlStyle: EMPTY_CLASS,
        OverlayView: EMPTY_CLASS,
        Point: EMPTY_CLASS,
        Polygon: EMPTY_CLASS,
        Polyline: EMPTY_CLASS,
        Rectangle: EMPTY_CLASS,
        SaveWidget: EMPTY_CLASS,
        ScaleControlStyle: EMPTY_CLASS,
        Size: EMPTY_CLASS,
        StreetViewCoverageLayer: EMPTY_CLASS,
        StreetViewPanorama: EMPTY_CLASS,
        StreetViewService: EMPTY_CLASS,
        StreetViewStatus: EMPTY_CLASS,
        StrokePosition: EMPTY_CLASS,
        StyledMapType: EMPTY_CLASS,
        SymbolPath: EMPTY_CLASS,
        TrafficLayer: EMPTY_CLASS,
        TransitLayer: EMPTY_CLASS,
        TransitMode: EMPTY_CLASS,
        TransitRoutePreference: EMPTY_CLASS,
        TravelMode: EMPTY_CLASS,
        UnitSystem: EMPTY_CLASS,
        ZoomControlStyle: EMPTY_CLASS,

        places: {
            AutocompleteService: class {
                getPlacePredictions(req, cb) {
                    getPlacePredictionsMock(req, cb);
                }
            },
            PlacesServiceStatus: {
                INVALID_REQUEST: 'INVALID_REQUEST',
                NOT_FOUND: 'NOT_FOUND',
                OK: 'OK',
                OVER_QUERY_LIMIT: 'OVER_QUERY_LIMIT',
                REQUEST_DENIED: 'REQUEST_DENIED',
                UNKNOWN_ERROR: 'UNKNOWN_ERROR',
                ZERO_RESULTS: 'ZERO_RESULTS',
            },
            PlacesAutocomplete: {
                INVALID_REQUEST: 'INVALID_REQUEST',
                NOT_FOUND: 'NOT_FOUND',
                OK: 'OK',
                OVER_QUERY_LIMIT: 'OVER_QUERY_LIMIT',
                REQUEST_DENIED: 'REQUEST_DENIED',
                UNKNOWN_ERROR: 'UNKNOWN_ERROR',
                ZERO_RESULTS: 'ZERO_RESULTS',
            },
        },
        Geocoder: class {},
        GeocoderStatus: {
            ERROR: 'ERROR',
            INVALID_REQUEST: 'INVALID_REQUEST',
            OK: 'OK',
            OVER_QUERY_LIMIT: 'OVER_QUERY_LIMIT',
            REQUEST_DENIED: 'REQUEST_DENIED',
            UNKNOWN_ERROR: 'UNKNOWN_ERROR',
            ZERO_RESULTS: 'ZERO_RESULTS',
        },
    },
};

export default () => {
    global.window.google = google;
};
