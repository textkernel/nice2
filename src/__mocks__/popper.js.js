import PopperJs from 'popper.js';

// eslint-disable-next-line import/no-default-export
export default class Popper {
    static placements = PopperJs.placements;

    constructor() {
        return {
            destroy: () => {},
            scheduleUpdate: () => {},
        };
    }
}
