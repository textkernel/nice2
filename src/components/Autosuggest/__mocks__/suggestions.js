export const SUGGESTIONS = [
    { name: 'Sun' },
    { name: 'Moon' },
    { name: 'Mercury' },
    { name: 'Venus' },
    { name: 'Earth' },
    { name: 'Mars' },
    { name: 'Jupiter' },
    { name: 'Saturn' },
    { name: 'Neptun' }
];

export const SUGGESTION_TO_STRING = item => (item ? item.name : '');
