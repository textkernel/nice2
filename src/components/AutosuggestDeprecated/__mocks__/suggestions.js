export const SUGGESTIONS = [
    { name: 'Sun' },
    { name: 'Moon' },
    { name: 'Mercury' },
    { name: 'Venus' },
    { name: 'Earth' },
    { name: 'Mars' },
    { name: 'Jupiter' },
    { name: 'Saturn' },
    { name: 'Neptun' },
];

export const COMPLEX_SUGGESTIONS = [
    { name: 'Sun', type: 'Star', count: 1234 },
    { name: 'Moon', type: 'Star', count: 1234 },
    { name: 'Mercury', type: 'Planet', count: 1234 },
    { name: 'Venus', type: 'Planet', count: 1234 },
    { name: 'Earth', type: 'Planet', count: 1234 },
    { name: 'Mars', type: 'Planet', count: 1234 },
    { name: 'Jupiter', type: 'Planet', count: 1234 },
    { name: 'Saturn', type: 'Planet', count: 1234 },
    { name: 'Neptun', type: 'Planet', count: 1234 },
];

export const SUGGESTION_TO_STRING = (item) => (item ? item.name : '');
