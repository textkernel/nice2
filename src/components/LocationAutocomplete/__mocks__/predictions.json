[
    {
        "description": "Truro, UK",
        "id": "51b6747d060368917397ce3cc25abe2089302262",
        "matched_substrings": [
            {
                "length": 2,
                "offset": 0
            }
        ],
        "place_id": "ChIJdRpa1XwQa0gRtAcdle9HY2E",
        "reference": "ChIJdRpa1XwQa0gRtAcdle9HY2E",
        "structured_formatting": {
            "main_text": "Truro",
            "main_text_matched_substrings": [
                {
                    "length": 2,
                    "offset": 0
                }
            ],
            "secondary_text": "UK"
        },
        "terms": [
            {
                "offset": 0,
                "value": "Truro"
            },
            {
                "offset": 7,
                "value": "UK"
            }
        ],
        "types": ["locality", "political", "geocode"]
    },
    {
        "description": "Trowbridge, UK",
        "id": "b10d478a406230745e98edd2d2a01f59097d5a59",
        "matched_substrings": [
            {
                "length": 2,
                "offset": 0
            }
        ],
        "place_id": "ChIJM6W0Ewp9cUgRLXIknC4a-Pw",
        "reference": "ChIJM6W0Ewp9cUgRLXIknC4a-Pw",
        "structured_formatting": {
            "main_text": "Trowbridge",
            "main_text_matched_substrings": [
                {
                    "length": 2,
                    "offset": 0
                }
            ],
            "secondary_text": "UK"
        },
        "terms": [
            {
                "offset": 0,
                "value": "Trowbridge"
            },
            {
                "offset": 12,
                "value": "UK"
            }
        ],
        "types": ["locality", "political", "geocode"]
    },
    {
        "description": "Tring, UK",
        "id": "8b1f6899d3555c6aae3a2d383ed96c7649f3ccf8",
        "matched_substrings": [
            {
                "length": 2,
                "offset": 0
            }
        ],
        "place_id": "ChIJ33SmFitYdkgR3-ErzH_p6Bw",
        "reference": "ChIJ33SmFitYdkgR3-ErzH_p6Bw",
        "structured_formatting": {
            "main_text": "Tring",
            "main_text_matched_substrings": [
                {
                    "length": 2,
                    "offset": 0
                }
            ],
            "secondary_text": "UK"
        },
        "terms": [
            {
                "offset": 0,
                "value": "Tring"
            },
            {
                "offset": 7,
                "value": "UK"
            }
        ],
        "types": ["locality", "political", "geocode"]
    },
    {
        "description": "Troon, UK",
        "id": "c563f4c69998a376dac8e3b4122067d782e752f7",
        "matched_substrings": [
            {
                "length": 2,
                "offset": 0
            }
        ],
        "place_id": "ChIJxy8qU3jRiUgR-sKnFLHxiSM",
        "reference": "ChIJxy8qU3jRiUgR-sKnFLHxiSM",
        "structured_formatting": {
            "main_text": "Troon",
            "main_text_matched_substrings": [
                {
                    "length": 2,
                    "offset": 0
                }
            ],
            "secondary_text": "UK"
        },
        "terms": [
            {
                "offset": 0,
                "value": "Troon"
            },
            {
                "offset": 7,
                "value": "UK"
            }
        ],
        "types": ["locality", "political", "geocode"]
    },
    {
        "description": "Tredegar, UK",
        "id": "4dca94194f54dee0879b9d368379eb5d09ec838d",
        "matched_substrings": [
            {
                "length": 2,
                "offset": 0
            }
        ],
        "place_id": "ChIJdQjXYyYkbkgRTRfZ8WAt6lI",
        "reference": "ChIJdQjXYyYkbkgRTRfZ8WAt6lI",
        "structured_formatting": {
            "main_text": "Tredegar",
            "main_text_matched_substrings": [
                {
                    "length": 2,
                    "offset": 0
                }
            ],
            "secondary_text": "UK"
        },
        "terms": [
            {
                "offset": 0,
                "value": "Tredegar"
            },
            {
                "offset": 10,
                "value": "UK"
            }
        ],
        "types": ["locality", "political", "geocode"]
    }
]
