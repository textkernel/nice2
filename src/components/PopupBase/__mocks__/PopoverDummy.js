import * as React from 'react';
import PropTypes from 'prop-types';
import { Button } from '../../..';

const PopoverDummy = React.forwardRef((props, ref) => {
    const { setPopupVisibility } = props;
    const popupStyle = {
        backgroundColor: '#FFEEDD',
        padding: '10px',
        borderRadius: '5px',
        border: '1px solid #CCC',
        boxShadow: '0 3px 6px 6px rgba(0, 0, 0, 0.05)',
    };
    return (
        <div style={popupStyle} ref={ref}>
            <p>I am the content of pop up.</p>
            <p>Can be a form or anything.</p>
            <Button onClick={() => setPopupVisibility(false)}>Close me!</Button>
        </div>
    );
});

PopoverDummy.propTypes = {
    setPopupVisibility: PropTypes.func.isRequired,
};

PopoverDummy.displayName = 'Popover';

export default PopoverDummy;
