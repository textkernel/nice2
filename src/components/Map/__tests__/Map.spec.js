import React from 'react';
import toJson from 'enzyme-to-json';
import {
    fitBoundsMock,
    setZoomMock,
    setCenterMock,
    geocodeMock,
    stabGoogleApi,
} from '../../../__mocks__/googleApiMock';
import { geocodeResponse } from '../__mocks__/geocodeResponse';
import { Map } from '..';

stabGoogleApi();

describe('<Map/> that renders a Map with markers', () => {
    const regionMarker = {
        center: {
            lat: 52.3922288,
            lng: 4.9338793,
        },
        radius: 30000,
    };
    const pointMarker = {
        center: {
            lat: 52.5112671,
            lng: 7.2535521,
        },
    };

    it('should render with default props', () => {
        const wrapper = mount(<Map />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('should set center and zoom when rendered with default props', () => {
        mount(<Map />);
        expect(setZoomMock).toHaveBeenCalledTimes(1);
        expect(setCenterMock).toHaveBeenCalledTimes(1);
    });
    it('should fit bounds when rendered with defaultArea as address', () => {
        geocodeMock.mockImplementationOnce((req, cb) => {
            cb(geocodeResponse.results, geocodeResponse.status);
        });
        mount(<Map defaultArea={{ address: 'nl' }} />);
        expect(geocodeMock).toHaveBeenCalledTimes(1);
        expect(fitBoundsMock).toHaveBeenCalledTimes(1);
    });
    it('should render with markers', () => {
        const wrapper = mount(<Map markers={[pointMarker, regionMarker]} />);
        expect(wrapper.find('Marker')).toHaveLength(2);
        expect(wrapper.find('Circle')).toHaveLength(1);
    });
    it('should fit map to markers', () => {
        mount(<Map markers={[pointMarker, regionMarker]} />);
        // call fitBounds for each marker
        expect(fitBoundsMock).toHaveBeenCalledTimes(2);
    });
    it('should fit map when new marker is added', () => {
        const wrapper = mount(<Map markers={[pointMarker]} />);
        expect(fitBoundsMock).toHaveBeenCalledTimes(1);
        wrapper.setProps({ markers: [pointMarker, regionMarker] });
        // call count = original call + again for each marker
        expect(fitBoundsMock).toHaveBeenCalledTimes(3);
    });
    it('should fit to default center and zoom if markers removed', () => {
        const wrapper = mount(<Map markers={[pointMarker, regionMarker]} />);
        wrapper.setProps({ markers: [] });
        expect(setZoomMock).toHaveBeenCalledTimes(1);
        expect(setCenterMock).toHaveBeenCalledTimes(1);
    });
});
