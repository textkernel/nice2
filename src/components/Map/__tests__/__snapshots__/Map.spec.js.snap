// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MapRendered/> that renders a Map with markers should render with default props 1`] = `
<Map
  center={
    Object {
      "lat": 52.3922288,
      "lng": 4.9338793,
    }
  }
  mapContainerStyle={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
  markers={Array []}
  zoom={7}
>
  <GoogleMap
    center={
      Object {
        "lat": 52.3922288,
        "lng": 4.9338793,
      }
    }
    mapContainerStyle={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
    onLoad={[Function]}
    options={
      Object {
        "fullscreenControl": false,
        "mapTypeControl": false,
        "rotateControl": false,
        "streetViewControl": false,
      }
    }
    zoom={7}
  >
    <div
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </GoogleMap>
</Map>
`;
